controller_manager/src/controller_manager.cpp

```diff
@@ -29,6 +29,18 @@ namespace controller_manager
 static constexpr const char * kControllerInterfaceName = "controller_interface";
 static constexpr const char * kControllerInterface = "controller_interface::ControllerInterface";
 
+// Changed services history QOS to keep all so we don't lose any client service calls
+static const rmw_qos_profile_t rmw_qos_profile_services_hist_keep_all = {
+  RMW_QOS_POLICY_HISTORY_KEEP_ALL,
+  1,  // message queue depth
```

Comment:
so I take it that `RMW_QOS_POLICY_HISTORY_KEEP_ALL` will render whatever is the message queue depth to be irrelevant? or why setting it to `1` below?

Reply:
I guess so. That's what I understood. We have to set it to something. And this setup definitely works with 10+ controllers.

Reply:
Yes, the message queue depth is ignored for 'keep all'. It might be good to include a comment indicating that.